### PR TITLE
add support for uniform body forces for IB structures

### PR DESCRIPTION
**Pull request by boyceg**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [X] Was clang-format run?
- [X] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [X] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [X] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: include/ibamr/IBStandardForceGen.h [hunk 1]
```diff
@@ -92,6 +92,12 @@ class IBStandardForceGen : public IBLagrangianForceStrategy
                                      const SpringForceFcnPtr spring_force_fcn_ptr,
                                      const SpringForceDerivFcnPtr spring_force_deriv_fcn_ptr = nullptr);
 
+    /*!
+     * \brief Set a uniform body force that is applied on each point in the
+     * structure with the given structure_id.
+     */
+    void setUniformBodyForce(IBTK::Vector F, int structure_id, int level_number);
```
Comment (abarret) [inline include/ibamr/IBStandardForceGen.h:99]: Should we also include an interface to set a variable body force by registering a function pointer? I can add this if you want me too.

Comment (boyceg) [reply include/ibamr/IBStandardForceGen.h:99]: If you want to add this, that would be good. Could use it for the uniform case too.

Comment (boyceg) [reply include/ibamr/IBStandardForceGen.h:99]: BTW, is it useful to have the interface for a uniform force, even if it is implemented internally using a function pointer? If so, could we go ahead and merge this (if the interface is OK) and then add the function pointer later?

Comment (abarret) [reply include/ibamr/IBStandardForceGen.h:99]: Sure. I'll create an issue.

**Review by abarret — APPROVED** [review]

**Comment by drwells** [comment]
We are missing a few things in the checklist. Do you plan on adding a test?

**Comment by boyceg** [comment]
Oh yeah, I will fill out the checklist! 😀

**Comment by boyceg** [comment]
@abarret, @drwells, can you take another look? I think this is ready to merge.

**Comment by drwells** [comment]
Nice. Could I convince you to update the CMakeLists.txt file for the tests?

**Comment by boyceg** [comment]
I think I did it.

**Comment by boyceg** [comment]
Hrmmm, I seem to have messed up the CMakeLists.txt file.

**Comment by boyceg** [comment]
OK, I think I have fixed CMakeLists.txt.

File: .gitignore [hunk 2]
```diff
@@ -8,7 +8,8 @@ ibtk/contrib/autotools/share
 ### Generated project files
 #
 # To regenerate: intercept-build --override-compiler make CC=intercept-cc CXX=intercept-c++
-*compile_commands*.json
+*compile_commands*.json*
+compile_flags.txt
```
Comment (abarret) [inline .gitignore:12]: This conflicts with #1264 
```suggestion

```

**Review by abarret — APPROVED** [review]
Looks fine to me.

**Pull request merged**